### ISSUE TITLE
[Bug]: zitadel ready command does not override default config parameters, causing the health check to fail.

### ISSUE THREAD

**Issue by everesio** [zitadel/zitadel#8106]
### Preflight Checklist

- [X] I could not find a solution in the documentation, the existing issues or discussions
- [ ] I have joined the [ZITADEL chat](https://zitadel.com/chat)

### Environment

Self-hosted

### Version

v2.54.1

### Database

PostgreSQL

### Database Version

16-alpine

### Describe the problem caused by this bug

zitadel config sets following parameters

```
ExternalPort: 9080
ExternalSecure: false
```
docker compose healthcheck is used as

```
healthcheck:
  test: ["CMD", "/app/zitadel", "ready", "--config", "/zitadel.yaml" ]
  interval: '10s'
  timeout: '5s'
  retries: 5
  start_period: '10s'
``` 

this healthcheck fails as default port 8080 and proto http is used (value are not overwritten)

```
Get \"https://localhost:8080/debug/ready\": http: server gave HTTP response to HTTPS client
``` 

### To reproduce

config zitadel.yaml

```
Log:
  Level: 'info'

ExternalPort: 9080
ExternalSecure: false
```

docker compose

``` 
services:
  zitadel:
    restart: always
    container_name: zitadel
    image: "ghcr.io/zitadel/zitadel:v2.54.1"
    command: 'start-from-init --config /zitadel.yaml --steps /zitadel.yaml --masterkey "MasterkeyNeedsToHave32Characters" --tlsMode disabled'
    ports:
      - "9080:8080"
    volumes:
      - ./zitadel.yaml:/zitadel.yaml:ro
    healthcheck:
      test: ["CMD", "/app/zitadel", "ready", "--config", "/zitadel.yaml" ]
      interval: '10s'
      timeout: '5s'
      retries: 5
      start_period: '10s'
    depends_on:
      db:
        condition: 'service_healthy'
  db:
    image: postgres:16-alpine
    restart: always
    container_name: zitadel-postgres
    environment:
      - POSTGRES_USER=root
      - POSTGRES_PASSWORD=postgres
    ports:
      - '5432:5432'
    healthcheck:
      test: ["CMD-SHELL", "pg_isready", "-d", "db_prod"]
      interval: 10s
      timeout: 60s
      retries: 5
      start_period: 10s
```
after started

``` 
docker inspect --format "{{json .State.Health }}" zitadel | jq '.Log[].Output'
docker exec -it zitadel /app/zitadel ready --config /zitadel.yaml
```

**Comment by doncicuto** [comment]
Hi @everesio, this issue is tricky. I'd say it's not a bug imho but this deserves a discussion which has been introduced in the past: https://github.com/zitadel/zitadel/issues/6863

The ready command may indeed get a config, but it ignores ExternalPort and ExternalSecure as those settings are related with configuring Zitadel behind a reverse proxy and does not affect the ready command which deals with Zitadel readiness no matter if a reverse proxy is set. So it's expected that your 9080 port is ignored. Here's the code that shows that the ready command ignores external settings as expected:

```
scheme := "https"
if !config.TLS.Enabled {
	scheme = "http"
}
// Checking the TLS cert is not in the scope of the readiness check
httpClient := http.Client{Transport: &http.Transport{TLSClientConfig: &tls.Config{InsecureSkipVerify: true}}}
res, err := httpClient.Get(scheme + "://" + net.JoinHostPort("localhost", strconv.Itoa(int(config.Port))) + "/debug/ready")
```

One thing it's the ready command and another one is the ready endpoint which is affected by the ExternalPort and ExternalSecure settings and the endpoint do indeed listen according to those settings.

For the TLS issue, if you use your custom zitadel.yaml the ready command will asume that the TLS config is enabled because that it's the default config for Zitadel and it doesn't know that you've set tlsMode disabled for your Zitadel instance. Therefore the ready command uses TLS for the client while the Zitadel instance is not and you get: `server gave HTTP response to HTTPS client`

If you add the following lines to your zitadel.yaml the ready command will work fine for your docker compose file

```
TLS:
  Enabled: false
```

In https://github.com/zitadel/zitadel/tree/main/e2e/config/localhost you can check an example where that TLS setting is added to zitadel.yaml and --tlsMode disabled is also used.

In summary I think this is not a bug but some clarification should be added to docs as @mffap proposed in https://github.com/zitadel/zitadel/issues/6863 

Maybe @eliobischof, @adlerhurst or @hifabienne can confirm if this should be treated as a bug or a docs issue that can be treated instead as a docs improvement

Thanks!

**Comment by everesio** [comment]
@doncicuto  thank you. I see the problem. 
If you want please close the ticket or use it to improve your documentation.

**Comment by eliobischof** [comment]
The compose health checks will be obsolete with #8684

**Comment by elinashoko** [comment]
Thanks for filing this issue. As part of an ongoing backlog cleanup, we're closing issues that have had no activity in over 12 months.

If this is still relevant, let us know in a comment (or reopen the issue) and we'll take another look.

This issue will close in 30 days if there's no further activity.
